Microsoft.Toolkit/Extensions/StringExtensions.cs

```diff
@@ -145,6 +146,7 @@ public static string FixHtml(this string html)
         /// <param name="format">The format of the string being linked.</param>
         /// <param name="args">The object which will receive the linked String.</param>
         /// <returns>Truncated string.</returns>
+        [Obsolete("This method will be removed in a future version of the Toolkit. Use the native C# string interpolation syntax instead, see: https://docs.microsoft.com/dotnet/csharp/language-reference/tokens/interpolated")]
         public static string AsFormat(this string format, params object[] args) => string.Format(format, args);
```

Comment:
I think I had added this for x:Bind, does String Interopolation work there?

Reply:
Oh, I see. So, it **should** work, and it does, but it's currently bugged (see https://github.com/microsoft/microsoft-ui-xaml/issues/2654).

For instance, this works fine:
```xml
<TextBlock Text="{x:Bind x:String.Format('Name: {0}', ViewModel.Name)}" />
```

But using more arguments results in a build-time error from the XAML compiler because it's doing an incorrect overload resolution (basically it's just picking whatever the first overload is alphabetically), so it ends up picking the one taking an `IFormatProvider` instead of a `string`, and then failing to build 😟

Let me know how you'd like to proceed, I'm fine with closing this if you think the extension should remain there just for this XAML bit. I wish this was just fixed on the XAML compiler end, as the current overload resolution is really not great.

Reply:
So, will this helper work around this bug or face the same issue?

If it helps get around the issue, then maybe let's leave it for now until we know that we can get a fix in WinUI 3?

Reply:
Well this is a bit annoying, I just checked with this extension and it seems the XAML compiler still chokes up anyway because it doesn't account for a `params T[]` parameter allowing multiple inputs. As in, this fails too:

```xml
<TextBlock Text="{x:Bind wct:StringExtensions.AsFormat('{0} {1}', ViewModel.Name, ViewModel.Surname)}"/>
```

With this error:

> Invalid binding path 'wct:StringExtensions.AsFormat('{0} {1}', ViewModel.Name, ViewModel.Surname)' : Missmatched function parameter count.

It just works with a single parameter basically, just with `string.Format` as usual. It seems to me due to the quirks of the XAML compiler we can then basically deprecate the extension without losing any actual functionality? 🤔

Reply:
@Sergio0694 yeah, let's deprecate, but point to the WinUI issue for reference.

Reply:
Not sure where to add the link to the WinUI issue, since that's XAML specific and the `Microsoft.Toolkit` is platform agnostic instead. Do you want me to just leave a link here in the PR or should I add that to the XML docs of the extension anyway?

Reply:
Maybe just in the code then vs. anything publicly visible? Just for our own knowledge within the toolkit?

Reply:
Sure, done in 1fb8c50574526dee9e458ae17834bced15916c2b 🙂